sensu/plugins/check-vyatta-tunnels.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+#
+# Checks that all tunnels are up
+#
+# Return CRITICAL if any tunnel is down
+#
+# Dean Daskalantonakis <ddaskal@us.ibm.com>
+# Carl Hannusch <channusch@us.ibm.com>
+
+import argparse
+import re
+import subprocess
+import sys
+
+STATE_OK = 0
+STATE_CRITICAL = 2
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-ip', '--ip', help='vyatta vrrp ip',
+                        required=True)
+    parser.add_argument('-p', '--password', help='snmp password',
+                        required=True)
+    args = parser.parse_args()
+
+    status = ""
+    cmd = "snmpwalk -u sensu -A %s -a md5 -l authNoPriv %s \
+           1.3.6.1.2.1.2.2.1.2" % (args.password, args.ip)
+    try:
+        device_list = subprocess.check_output(cmd, shell=True).split("\n")
+    except subprocess.CalledProcessError as e:
+        print(e.output)
+        print('CRITICAL: Failed to retrieve Vyatta devices \
+                name via SNMP')
+        sys.exit(STATE_CRITICAL)
+
+    cmd = "snmpwalk -u sensu -A %s -a md5 -l authNoPriv %s \
+           1.3.6.1.2.1.2.2.1.8" % (args.password, args.ip)
+    try:
+        device_status = subprocess.check_output(cmd, shell=True).split("\n")
+    except subprocess.CalledProcessError as e:
+        print(e.output)
+        print('CRITICAL: Failed to retrieve Vyatta device states \
+                status via SNMP')
+        sys.exit(STATE_CRITICAL)
+
+    for index, device in enumerate(device_list):
+        if " vti" in str(device):
```

Comment:
yes, there is also an ip_vti in the list which does not give us the information that we need.